examples/constraints.rs

```diff
@@ -0,0 +1,139 @@
+
+        if let Event::Key(key) = event::read()? {
+            if let KeyCode::Char('q') = key.code {
+                return Ok(());
+            }
+            if let KeyCode::Char('j') = key.code {
+                selection += 1
+            }
+            if let KeyCode::Char('k') = key.code {
+                selection = selection.saturating_sub(1)
+            }
+        }
+        selection = selection.clamp(0, 6)
+    }
+}
+
+fn paragraph<T: Into<Text<'static>>>(text: T, bordered: bool) -> Paragraph<'static> {
+    if bordered {
+        Paragraph::new(text)
+            .alignment(Alignment::Center)
+            .block(Block::bordered().padding(Padding::new(0, 0, 0, 0)))
+    } else {
+        Paragraph::new(text).block(Block::default().padding(Padding::new(0, 0, 1, 0)))
+    }
+}
+
+fn example(f: &mut Frame, area: Rect, constraints: Vec<Constraint>, heading: String) {
+    let [title, area] = area.split(&Layout::vertical([Ratio(1, 2); 2]));
+    f.render_widget(paragraph(heading.to_string(), false), title);
+    let blocks = Layout::horizontal(constraints).split(area);
+    for b in blocks.iter() {
+        f.render_widget(paragraph(format!("{} px", b.width), true), *b);
+    }
+}
+
+macro_rules! example {
+    ($f:ident, $area:ident, [$($args:tt)*]) => {
+        example($f, $area, vec![$($args)*], format!("`{}`", stringify!([$($args)*])));
+    };
+}
+
+fn ui(f: &mut Frame, tab: usize) {
+    let [tabs, area] = f.size().split(&Layout::vertical([Fixed(3), Min(0)]));
+    f.render_widget(
+        Tabs::new([
+            "Fixed",
+            "Length",
+            "Percentage",
+            "Ratio",
+            "Fill",
+            "Min",
+            "Max",
+        ])
+        .block(Block::bordered().title("Constraints"))
+        .highlight_style(Style::default().yellow())
+        .select(tab)
+        .padding("  ", "  "),
+        tabs,
+    );
+    let [area, _] = area.split(&Layout::horizontal([Fixed(50), Min(0)]));
+    let [example1, example2, _] = area.split(&Layout::vertical([Fixed(6); 3]));
+
+    match tab {
+        0 => {
+            example!(f, example1, [Fixed(40), Min(0)]);
+            example!(f, example2, [Fixed(20), Fixed(20), Min(0)]);
+        }
+        1 => {
+            example!(f, example1, [Length(20), Fixed(20)]);
+            example!(f, example2, [Length(20), Length(20)]);
+        }
+
+        2 => {
+            example!(f, example1, [Percentage(75), Min(0)]);
+            example!(f, example2, [Percentage(25), Min(0)]);
+        }
+
+        3 => {
+            example!(f, example1, [Ratio(1, 2); 2]);
+            example!(f, example2, [Ratio(1, 4); 4]);
+        }
+
+        4 => {
+            example!(f, example1, [Fill(1), Fill(2), Fill(3)]);
+            example!(f, example2, [Fill(1), Percentage(50), Fill(1)]);
+        }
+
+        5 => {
+            example!(f, example1, [Percentage(100), Min(20)]);
+            example!(f, example2, [Percentage(100), Min(10)]);
```

Comment:
Do you think users would want to do that? 

That seems like it would not be well defined and might be one of those things that would change with weights in an odd way. I'm happy to add a test case for that.

Reply:
Not sure - it just seemed like a question to ask when I used the example and noticed that the examples showed 100%/min(0) and 0%/max(20).

Reply:
I've added these examples in there.